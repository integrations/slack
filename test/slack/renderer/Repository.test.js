const moment = require('moment');
const { Repository } = require('../../../lib/slack/renderer/repository');
const repoFixture = require('../../fixtures/repo.json');

repoFixture.updated_at = moment().subtract(2, 'months');

describe('Repository rendering', () => {
  test('works for full unfurls', async () => {
    const repositoryMessage = new Repository({
      repository: repoFixture,
      unfurlType: 'full',
    });
    expect(repositoryMessage.getRenderedMessage()).toMatchSnapshot();
  });

  test('works for condensed unfurls', async () => {
    const repositoryMessage = new Repository({
      repository: repoFixture,
      unfurlType: 'condensed',
    });
    expect(repositoryMessage.getRenderedMessage()).toMatchSnapshot();
  });
});
