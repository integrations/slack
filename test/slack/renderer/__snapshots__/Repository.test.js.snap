// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository rendering works 1`] = `
Object {
  "color": "#24292f",
  "fallback": "bkeepers/dotenv",
  "fields": Array [
    Object {
      "short": true,
      "title": "Watchers",
      "value": 57,
    },
    Object {
      "short": true,
      "title": "Stars",
      "value": 4230,
    },
    Object {
      "short": true,
      "title": "Forks",
      "value": 283,
    },
    Object {
      "short": true,
      "title": "Last updated",
      "value": "2 months ago",
    },
  ],
  "footer": "<https://github.com/bkeepers/dotenv|View it on GitHub>",
  "footer_icon": "https://assets-cdn.github.com/favicon.ico",
  "mrkdwn_in": Array [
    "text",
    "fields",
  ],
  "text": "A Ruby gem to load environment variables from \`.env\`. ",
  "title": "bkeepers/dotenv",
  "ts": 1343101131,
}
`;
